### ISSUE TITLE
no supervisor internet connection - On installation. 

### ISSUE THREAD

**Issue by RichardPar** [home-assistant/supervisor#5087]
### Describe the issue you are experiencing

Doing a new install on a Khadas VIM Debian12 - The installer does not progress - complaining about Internet. Is there a qualification on what is needed for internet to be detected correctly?

I can Ping, DNS etc etc all just fine.. Date and time is correct and NTP synced
```
root@debian12:~# ping 4.2.2.1
PING 4.2.2.1 (4.2.2.1) 56(84) bytes of data.
64 bytes from 4.2.2.1: icmp_seq=1 ttl=59 time=11.1 ms
64 bytes from 4.2.2.1: icmp_seq=2 ttl=59 time=11.1 ms
^C
--- 4.2.2.1 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1002ms
rtt min/avg/max/mdev = 11.057/11.064/11.071/0.007 ms
root@debian12:~#

root@debian12:~# ping www.sky.com
PING e1264.g.akamaiedge.net (23.43.76.120) 56(84) bytes of data.
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=1 ttl=57 time=10.9 ms
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=2 ttl=57 time=10.5 ms
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=3 ttl=57 time=11.4 ms
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=4 ttl=57 time=11.5 ms
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=5 ttl=57 time=11.3 ms
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=6 ttl=57 time=11.1 ms
64 bytes from a23-43-76-120.deploy.static.akamaitechnologies.com (23.43.76.120): icmp_seq=7 ttl=57 time=11.1 ms
^C
--- e1264.g.akamaiedge.net ping statistics ---
7 packets transmitted, 7 received, 0% packet loss, time 6010ms
rtt min/avg/max/mdev = 10.538/11.096/11.493/0.297 ms

```

Thanks
Richard

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Debian

### Steps to reproduce the issue

1.
2.
3.
...

### Anything in the Supervisor logs that might be useful for us?

```txt
HA fails to install
```

### System Health information

```
2024-05-17T12:22:35.731945+01:00 debian12 avahi-daemon[830]: Leaving mDNS multicast group on interface eth0.IPv6 with address fe80::154b:7c26:f590:7068.
2024-05-17T12:22:35.733192+01:00 debian12 avahi-daemon[830]: Joining mDNS multicast group on interface eth0.IPv6 with address fde5:5fc4:30d8:c02e:5ed8:f2af:62ff:7efa.
2024-05-17T12:22:35.733643+01:00 debian12 avahi-daemon[830]: Registering new address record for fde5:5fc4:30d8:c02e:5ed8:f2af:62ff:7efa on eth0.*.
2024-05-17T12:22:35.734069+01:00 debian12 avahi-daemon[830]: Withdrawing address record for fe80::154b:7c26:f590:7068 on eth0.
2024-05-17T12:22:38.731170+01:00 debian12 kernel: [   42.510247][0 T0     dh] cec: bad initiator with self 0x0
2024-05-17T12:22:38.883402+01:00 debian12 kernel: [   42.654076][0 T0     dh] cec: bad initiator with self 0x0
2024-05-17T12:22:39.050637+01:00 debian12 kernel: [   42.806196][0 T0     dh] cec: bad initiator with self 0x0
2024-05-17T12:22:39.122933+01:00 debian12 hassio_supervisor[1165]: #033[33m2024-05-17 12:22:39.109 WARNING (MainThread) [supervisor.jobs] 'Updater.fetch_data' blocked from execution, no su
pervisor internet connection#033[0m
2024-05-17T12:22:39.153291+01:00 debian12 hassio_supervisor[1165]: #033[32m2024-05-17 12:22:39.149 INFO (MainThread) [supervisor.docker.interface] Found None versions: [<AwesomeVersion Cal
Ver '2024.05.1'>, <AwesomeVersion SpecialContainer 'latest'>]#033[0m
2024-05-17T12:22:39.173325+01:00 debian12 hassio_supervisor[1165]: #033[32m2024-05-17 12:22:39.168 INFO (MainThread) [supervisor.plugins.base] No cli plugin Docker image None found.#033[0m
2024-05-17T12:22:39.175016+01:00 debian12 hassio_supervisor[1165]: #033[32m2024-05-17 12:22:39.170 INFO (MainThread) [supervisor.plugins.base] Setup cli plugin#033[0m
2024-05-17T12:22:44.194645+01:00 debian12 hassio_supervisor[1165]: #033[33m2024-05-17 12:22:44.187 WARNING (MainThread) [supervisor.jobs] 'Updater.fetch_data' blocked from execution, no su
pervisor internet connection#033[0m
2024-05-17T12:22:44.195749+01:00 debian12 hassio_supervisor[1165]: #033[33m2024-05-17 12:22:44.187 WARNING (MainThread) [supervisor.plugins.base] Error on installing cli plugin, retrying in 30sec

### Supervisor diagnostics

_No response_

### Additional information

None

**Comment by RichardPar** [comment]
Dont know if its relevant... but it adds 3 DNS servers - and then complains there are no DNS servers specified.

this normal?

```
2024-05-17T12:22:33.780461+01:00 debian12 avahi-daemon[830]: Joining mDNS multicast group on interface eth0.IPv4 with address 192.168.10.147.
2024-05-17T12:22:33.780828+01:00 debian12 systemd-resolved[751]: eth0: Bus client set MulticastDNS setting: yes
2024-05-17T12:22:33.781262+01:00 debian12 avahi-daemon[830]: New relevant interface eth0.IPv4 for mDNS.
2024-05-17T12:22:33.781491+01:00 debian12 avahi-daemon[830]: Registering new address record for 192.168.10.147 on eth0.IPv4.
2024-05-17T12:22:33.784133+01:00 debian12 systemd-resolved[751]: eth0: Bus client set default route setting: yes
2024-05-17T12:22:33.793294+01:00 debian12 systemd-resolved[751]: eth0: Bus client set DNS server list to: 192.168.10.254, 4.2.2.1, 8.8.8.8
2024-05-17T12:22:33.800534+01:00 debian12 NetworkManager[887]: <info>  [1715944953.7958] device (eth0): state change: ip-config -> ip-check (reason 'none', sys-iface-state: 'managed')
2024-05-17T12:22:33.802653+01:00 debian12 NetworkManager[887]: <info>  [1715944953.8022] device (eth0): state change: ip-check -> secondaries (reason 'none', sys-iface-state: 'managed')
2024-05-17T12:22:33.812634+01:00 debian12 NetworkManager[887]: <info>  [1715944953.8030] device (eth0): state change: secondaries -> activated (reason 'none', sys-iface-state: 'managed')
2024-05-17T12:22:33.813255+01:00 debian12 NetworkManager[887]: <info>  [1715944953.8043] manager: NetworkManager state is now CONNECTED_SITE
2024-05-17T12:22:33.813723+01:00 debian12 NetworkManager[887]: <info>  [1715944953.8053] device (eth0): Activation: successful, device activated.
2024-05-17T12:22:33.814021+01:00 debian12 hassio_supervisor[1165]: #033[32m2024-05-17 12:22:33.811 INFO (MainThread) [supervisor.host.network] Updating local network information#033[0m
2024-05-17T12:22:33.818529+01:00 debian12 systemd-resolved[751]: Using degraded feature set UDP instead of UDP+EDNS0 for DNS server 192.168.10.254.
2024-05-17T12:22:33.851320+01:00 debian12 NetworkManager[887]: <info>  [1715944953.8510] manager: NetworkManager state is now CONNECTED_GLOBAL
2024-05-17T12:22:33.970569+01:00 debian12 nm-dispatcher[1776]: Dropped protocol specifier '.inet' from 'eth0.inet'. Using 'eth0' (ifindex=3).
2024-05-17T12:22:33.970738+01:00 debian12 nm-dispatcher[1776]: No DNS servers specified, refusing operation.
2024-05-17T12:22:34.073661+01:00 debian12 hassio_supervisor[1165]: #033[32m2024-05-17 12:22:34.070 INFO (MainThread) [supervisor.host.apparmor] Loading AppArmor Profiles: {'hassio-supervisor'}#033[0m
2024-05-17T12:22:34.074542+01:00 debian12 os-agent[814]: INFO: 2024/05/17 12:22:34 apparmor.go:45: Load AppArmor profile '/usr/share/hassio/apparmor/hassio-supervisor'.
2024-05-17T12:22:34.081438+01:00 debian12 nm-dispatcher[1818]: Dropped protocol specifier '.inet6' from 'eth0.inet6'. Using 'eth0' (ifindex=3).
2024-05-17T12:22:34.081715+01:00 debian12 nm-dispatcher[1818]: No DNS servers specified, refusing operation.
2024-05-17T12:22:34.091482+01:00 debian12 os-agent[814]: INFO: 2024/05/17 12:22:34 apparmor.go:53: Load profile '/usr/share/hassio/apparmor/hassio-supervisor':
```

**Comment by RichardPar** [comment]
UPDATE : The hassio_supervisor container is not getting the DNS server entries for some reason

**Comment by agners** [comment]
This should be addressed with #5088.
